pkg/scheduler/queue/queue.go

```diff
@@ -239,6 +266,8 @@ func (q *RequestQueue) dispatcherLoop() {
 		case <-q.stopRequested:
 			// Nothing much to do here - fall through to the stop logic below to see if we can stop immediately.
 			stopping = true
+		case <-q.observeInflightRequests:
```

Comment:
We need a dedicated goroutine just to track it at fixed regular intervals (keep it simple, using a timer, not publishing messages through a chan). The problem of doing it here is that the frequency is not guaranteed, because the `for` loop could be busy doing other stuff.

Reply:
OK Charles had requested I avoid atomics , but I will switch back to the atomics I had used originally